diff --git a/CRM/Core/Payment/iATSService.php b/CRM/Core/Payment/iATSService.php
index 149b670..e870f20 100644
--- a/CRM/Core/Payment/iATSService.php
+++ b/CRM/Core/Payment/iATSService.php
@@ -129,6 +129,7 @@ class CRM_Core_Payment_iATSService extends CRM_Core_Payment {
    */
   public function doPayment(&$params, $component = 'contribute') {
 
+    CRM_Utils_Hook::alterPaymentProcessorParams($this, $params, $params);
     if (!$this->_profile) {
       return self::error('Unexpected error, missing profile');
     }
diff --git a/CRM/Core/Payment/iATSServiceACHEFT.php b/CRM/Core/Payment/iATSServiceACHEFT.php
index e9b5f63..1dc86b7 100644
--- a/CRM/Core/Payment/iATSServiceACHEFT.php
+++ b/CRM/Core/Payment/iATSServiceACHEFT.php
@@ -203,6 +203,7 @@ class CRM_Core_Payment_iATSServiceACHEFT extends CRM_Core_Payment_iATSService {
    */
   public function doPayment(&$params, $component = 'contribute') {
 
+    CRM_Utils_Hook::alterPaymentProcessorParams($this, $params, $params);
     if (!$this->_profile) {
       return self::error('Unexpected error, missing profile');
     }
diff --git a/js/recur_start.js b/js/recur_start.js
index 1260b7a..7ebb268 100644
--- a/js/recur_start.js
+++ b/js/recur_start.js
@@ -6,27 +6,50 @@
 /*global CRM, ts */
 cj(function ($) {
   'use strict';
-   if ($('.is_recur-section').length) {
-     $('.is_recur-section #iats-recurring-start-date').remove();
-     $('.is_recur-section').append($('#iats-recurring-start-date'));
-     cj('input[id="is_recur"]').on('change', function() {
-       toggleRecur();
-     });
-     toggleRecur();
-   }
-   else { // I'm not on the right kind of page, just remove the extra field
-     $('#iats-recurring-start-date').remove();
-   }
+  $(document).ready(function() {
+    if ($('.is_recur-section').length) {
+      $('.is_recur-section #iats-recurring-start-date').remove();
+      $('.is_recur-section').append($('#iats-recurring-start-date'));
+      cj('input[id="is_recur"]').on('change', function() {
+        toggleRecur();
+      });
+      toggleRecur();
+    }
+    else {
+      if (!$('.delayedFields-section').length) {
+        // I'm not on the right kind of page, just remove the extra field
+        $('#iats-recurring-start-date').remove();
+      }
+      else {
+        cj('#partial_payment').on('change', function() {
+          toggleRecurringStartDate();
+        });
+        toggleRecurringStartDate();
+      }
+    }
+  });
 
-   function toggleRecur( ) {
-     var isRecur = $('input[id="is_recur"]:checked');
-     if (isRecur.val() > 0) {
-       $('#iats-recurring-start-date').show().val('');
-     }
-     else {
-       $('#iats-recurring-start-date').hide();
-       $("#iats-recurring-start-date option:selected").prop("selected", false);
-       $("#iats-recurring-start-date option:first").prop("selected", "selected");
-     }
-   }
+  function toggleRecurringStartDate( ) {
+    console.log($('#partial_payment').prop('checked'));
+    if ($('#partial_payment').prop('checked')) {
+      $('#iats-recurring-start-date').show().val('');
+    }
+    else {
+      $('#iats-recurring-start-date').hide();
+      $("#iats-recurring-start-date option:selected").prop("selected", false);
+      $("#iats-recurring-start-date option:first").prop("selected", "selected");
+    }
+  };
+
+  function toggleRecur( ) {
+    var isRecur = $('input[id="is_recur"]:checked');
+    if (isRecur.val() > 0) {
+      $('#iats-recurring-start-date').show().val('');
+    }
+    else {
+      $('#iats-recurring-start-date').hide();
+      $("#iats-recurring-start-date option:selected").prop("selected", false);
+      $("#iats-recurring-start-date option:first").prop("selected", "selected");
+    }
+  };
 });
